### PR TITLE
Align corner brackets to the top and bottom when scaling

### PR DESCRIPTION
**Pull request by bluebear94**
See #4188

This broke scaling of the corner brackets, U+231C – U+231F. Hopefully this didn’t happen to be needed? Right?

### CODE REVIEW

**Comment by PgBiel** [comment]
> This broke scaling of the corner brackets, U+231C – U+231F.

Could you provide some images comparing the old and the new behavior? Should make it easier to tell if that's a problem or not (e.g. if the previous behavior was unintended).

**Comment by Enter-tainer** [comment]
I think there are images in #4188 but i wonder if it's the correct place to apply "fix". And i'm also surprised to see non of other tests fail

**Comment by Enivex** [comment]
> I think there are images in #4188 but i wonder if it's the correct place to apply "fix". And i'm also surprised to see non of other tests fail

Presumably because these are some of the few delimiters that arent vertically symmetric.

**Comment by Enivex** [comment]
https://github.com/typst/typst/pull/1756

This was done purposely

**Comment by bluebear94** [comment]
Both LuaTeX and XeTeX have incorrect results with `\left` and `\right` (though less so with LuaTeX):

```tex
\usepackage{unicode-math}

\[\ulcorner c \urcorner \ulcorner \frac{a}{b} \urcorner\]

\[\left\ulcorner c \right\urcorner \left\ulcorner \frac{a}{b} \right\urcorner\]
```

gives the following in LuaLaTeX:

image

and the following in XeLaTeX:

image

**Comment by Enivex** [comment]
Okay, so I guess we'll have to come up with a solution ourselves.

**Comment by Enivex** [comment]
Can these specific delimiters be top and bottom aligned?

File: tests/suite/math/delimited.typ [hunk 1]
```diff
@@ -62,3 +62,7 @@ $ 1/(2 y (x) (2(3)) $
 // Test ignoring weak spacing immediately after the opening
 // and immediately before the closing.
 $ [#h(1em, weak: true)A(dif x, f(x) dif x)sum#h(1em, weak: true)] $
+
+--- math-lr-corner-brackets ---
+// Test positioning of U+231C to U+231F
```
Comment (laurmaedje) [inline tests/suite/math/delimited.typ:67]: could you add a comment here that they are supposed to look the same? the "vs" had me briefly thinking that they are supposed to look _different._

could also rename it to `issue-4188-lr-corner-brackets` to link it to that issue

File: crates/typst/src/math/fragment.rs [hunk 2]
```diff
@@ -408,9 +410,16 @@ impl VariantFragment {
...
+    }
+
+    /// Vertically adjust the fragment's frame so that it is aligned
+    /// to the given alignment on the axis.
+    pub fn align_on_axis(&mut self, ctx: &MathContext, align: VAlignment) {
         let h = self.frame.height();
         let axis = ctx.constants.axis_height().scaled(ctx, self.font_size);
-        self.frame.set_baseline(h / 2.0 + axis);
+        let align = (2 - align as u8) as f64;
+        self.frame.set_baseline((h / 2.0 + axis) * align);
```
Comment (laurmaedje) [inline crates/typst/src/math/fragment.rs:422]: can this use `align.position(..)`? The `(2 - align as u8)` is too clever for my liking and also creates an implicit dependence on the order of the enum.

File: tests/suite/math/delimited.typ (hunk 1 above)
Comment (bluebear94) [reply tests/suite/math/delimited.typ:67]: I’ve changed the `vs` to an `=`.

File: crates/typst/src/math/fragment.rs (hunk 2 above)
Comment (bluebear94) [reply crates/typst/src/math/fragment.rs]: This hasn’t been defined for `VAlignment` yet, but I’ve done this.

**Comment by laurmaedje** [comment]
Thanks!

**Comment by istudyatuni** [comment]
I guess linked issue could be closed

**Pull request merged**